### PR TITLE
Replace integer literals cast to bool with bool literals

### PR DESCRIPTION
**Pull request by thebrid**
## Description
I ran clang-tidy across the whole codebase checking for [modernize-use-bool-literals](https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-bool-literals.html). This PR fixes the detected errors.

## Motivation and Context
This looks nicer & saves a type conversion.

## How Has This Been Tested?
Ran existing tests, all passed.

## Screenshots (if appropriate):
N/A

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [X] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

File: xbmc/peripherals/devices/PeripheralCecAdapter.cpp [hunk 1]
```diff
@@ -1346,11 +1346,11 @@ void CPeripheralCecAdapter::SetConfigurationFromSettings(void)
     ReadLogicalAddresses(GetSettingInt("standby_devices"), m_configuration.powerOffDevices);
 
   // read the boolean settings
-  m_bUseTVMenuLanguage                 = GetSettingBool("use_tv_menu_language") ? 1 : 0;
+  m_bUseTVMenuLanguage                 = GetSettingBool("use_tv_menu_language") ? true : false;
```
Comment (garbear) [inline xbmc/peripherals/devices/PeripheralCecAdapter.cpp]: If GetSettingBool returns a bool, then the ternary isn't necessary

File: xbmc/dialogs/GUIDialogYesNo.cpp [hunk 2]
```diff
@@ -29,7 +29,7 @@ bool CGUIDialogYesNo::OnMessage(CGUIMessage& message)
     {
       int iControl = message.GetSenderId();
       int iAction = message.GetParam1();
-      if (1 || ACTION_SELECT_ITEM == iAction)
+      if (true || ACTION_SELECT_ITEM == iAction)
```
Comment (garbear) [inline xbmc/dialogs/GUIDialogYesNo.cpp]: Interesting. It seems this line hasn't been touched in over 10 years...

File: xbmc/peripherals/devices/PeripheralCecAdapter.cpp (hunk 1 above)
Comment (thebrid) [reply xbmc/peripherals/devices/PeripheralCecAdapter.cpp]: @garbear `clang-tidy` also has a [readability-simplify-boolean-expr](https://clang.llvm.org/extra/clang-tidy/checks/readability-simplify-boolean-expr.html) check which will cleanup expressions like these.

I'm planning to send a follow-up PR for that but I'll add a commit here to cleanup the instances you mentioned.

File: lib/libUPnP/Platinum/Source/Core/PltSsdp.cpp [hunk 3]
```diff
@@ -282,7 +282,7 @@ PLT_SsdpDeviceAnnounceTask::DoRun()
 {
     NPT_List<NPT_NetworkInterface*> if_list;
 
-    while (1) {
+    while (true) {
```
Comment (Rechi) [inline lib/libUPnP/Platinum/Source/Core/PltSsdp.cpp]: Please drop this change as it's not our library.

File: lib/libUPnP/Platinum/Source/Core/PltTaskManager.cpp [hunk 4]
```diff
@@ -127,7 +127,7 @@ PLT_TaskManager::Abort()
             break; 
         
         NPT_System::Sleep(NPT_TimeInterval(0.05));
-    } while (1);
+    } while (true);
```
Comment (Rechi) [inline lib/libUPnP/Platinum/Source/Core/PltTaskManager.cpp]: same

File: lib/libUPnP/Platinum/Source/Devices/MediaServer/PltSyncMediaBrowser.cpp [hunk 5]
```diff
@@ -466,7 +466,7 @@ PLT_SyncMediaBrowser::BrowseSync(PLT_DeviceDataReference&      device,
 
         // ask for the next chunk of entries
         index = count;
-    } while(1);
+    } while(true);
```
Comment (Rechi) [inline lib/libUPnP/Platinum/Source/Devices/MediaServer/PltSyncMediaBrowser.cpp]: same

File: lib/libUPnP/Platinum/Source/Devices/MediaServer/PltSyncMediaBrowser.cpp [hunk 6]
```diff
@@ -552,7 +552,7 @@ PLT_SyncMediaBrowser::SearchSync(PLT_DeviceDataReference&      device,
...
```
Comment (Rechi) [inline lib/libUPnP/Platinum/Source/Devices/MediaServer/PltSyncMediaBrowser.cpp]: same

File: xbmc/addons/PVRClient.cpp [hunk 7]
```diff
@@ -365,7 +365,7 @@ void CPVRClient::WriteClientChannelInfo(const CPVRChannelPtr &xbmcChannel, PVR_C
 bool CPVRClient::GetAddonProperties(void)
 {
   std::string strBackendName, strConnectionString, strFriendlyName, strBackendVersion, strBackendHostname;
-  PVR_ADDON_CAPABILITIES addonCapabilities = {0};
+  PVR_ADDON_CAPABILITIES addonCapabilities = {false};
```
Comment (Rechi) [inline xbmc/addons/PVRClient.cpp]: `PVR_ADDON_CAPABILITIES` struct has members which are not of type bool.

Comment (ksooo) [reply xbmc/addons/PVRClient.cpp]: Yep, please reverse this change.

File: CMakeLists.txt [hunk 8]
```diff
@@ -42,6 +42,7 @@ set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
 set(CMAKE_ASM_FLAGS "${CMAKE_C_FLAGS} -x assembler-with-cpp")
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
```
Comment (Rechi) [inline CMakeLists.txt]: This change is unrelated to this PR, please remove it.

**Comment by Rechi** [comment]
wrong revision was build
jenkins build this please
